src/demo/pages/Home.tsx

```diff
@@ -1,31 +1,31 @@
-import React, { useEffect } from "react";
-
-import { api, SdkError, SdkErrors, ExternalUser } from "@tonomy/tonomy-id-sdk";
+import React, { useContext, useEffect } from "react";
+import { api, SdkError, SdkErrors } from "@tonomy/tonomy-id-sdk";
 import settings from "../../common/settings";
 import "./Home.css";
-import { TH1, TH3, TP } from "../../common/atoms/THeadings";
+import { TP } from "../../common/atoms/THeadings";
 import logo from "/tonomy-logo48.png";
 import Rectangle from "../assets/Rectangle.png";
 import HandImage from "../assets/handImg.png";
-import { Highlighter } from "rc-highlight";
 import "@tonomy/tonomy-id-sdk/build/api/tonomy.css";
 import { useNavigate } from "react-router-dom";
 import useErrorStore from "../../common/stores/errorStore";
+import { AuthContext } from "../providers/AuthProvider";
 
 export default function Home() {
+  const { user } = useContext(AuthContext);
+
+  const navigation = useNavigate();
   const errorStore = useErrorStore();
 
   async function onButtonPress() {
     api.ExternalUser.loginWithTonomy({ callbackPath: "/callback" });
   }
 
-  const navigation = useNavigate();
-
   async function onRender() {
     try {
-      await api.ExternalUser.getUser({ autoLogout: false });
-      // User is logged in
-      navigation("/user-home");
+      const user = await api.ExternalUser.getUser({ autoLogout: false });
```

Comment:
this should not be with `getUser()` (otherwise it `getUser()` will be called twice unnecessarily), we should be able to see if it exists from the AuthProvider

Reply:
if we keep this then 

if the user tries to go to the "/" home page and are already logged in, it wil redirect to /user-home page. I didn't wrap the Home Page with Auth Provider so if user is not logged in it only call it once. if we get the user from AuthProvider it will make the user undefined if someone refresh the "/" page

Reply:
_I think_ AuthProvider gets injected at root level on top of ALL Routes. Like in https://github.com/remix-run/react-router/blob/dev/examples/auth/src/App.tsx

It should be that all routes can check if the user is logged in or not...

But this might be good enough for now. Lets have a quick chat when you're ready.